tests/test_util.py

```diff
@@ -402,25 +421,38 @@ def request_callback(request):
 
                 digest = 'v1-converted-digest'
 
-        headers = {
-            'Content-Type': '{}+jsonish'.format(media_type_prefix),
-            'Docker-Content-Digest': digest
-        }
+        headers = {}
+        if digest_is_v1 or has_content_type_header:
```

Comment:
The code you are pointing to seems to be the code to import docker images into pulp. Wouldn't the relevant code be what pulp-crane does? (And pulp-crane *could* return the digest since the v2 protocol specifies that the digest is returned with the redirect and not the final content.)